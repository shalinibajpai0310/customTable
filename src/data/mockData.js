export const mockState = {
  customers: [
    {
      _id: "customer_0",
      index: 0,
      age: 31,
      name: "Price Valdez",
      email: "pricevaldez@metroz.com",
      phone: "+1 (804) 410-3749",
      address: "789 Remsen Avenue, Washington, Louisiana, 7638",
      orders: [
        {
          id: 0,
          orderId: "010001",
          orderNumber: "010001",
          orderDate: "2019-09-01",
          items: [
            {
              _id: 0,
              itemId: "P01",
              itemName: "Pen",
              quantity: "2",
              price: "25"
            },
            {
              _id: 1,
              itemId: "PC01",
              itemName: "Pencil",
              quantity: "2",
              price: "6"
            },
            {
              _id: 2,
              itemId: "M01",
              itemName: "Marker",
              quantity: "2",
              price: "50"
            },
            {
              _id: 3,
              itemId: "D01",
              itemName: "Duster",
              quantity: "2",
              price: "100"
            }
          ]
        },
        {
          id: 1,
          orderId: "010002",
          orderNumber: "010002",
          orderDate: "2019-09-01",
          items: [
            {
              _id: 0,
              itemId: "S01",
              itemName: "Stickers",
              quantity: "2",
              price: "25"
            },
            {
              _id: 1,
              itemId: "B01",
              itemName: "Balloons",
              quantity: "2",
              price: "6"
            },
            {
              _id: 2,
              itemId: "RIB01",
              itemName: "Ribbons",
              quantity: "2",
              price: "50"
            },
            {
              _id: 3,
              itemId: "M01",
              itemName: "Marker",
              quantity: "2",
              price: "100"
            }
          ]
        },
        {
          id: 2,
          orderId: "010003",
          orderNumber: "010003",
          orderDate: "2019-09-01",
          items: [
            {
              _id: 0,
              itemId: "GL01",
              itemName: "Glue",
              quantity: "2",
              price: "25"
            },
            {
              _id: 1,
              itemId: "PC01",
              itemName: "Pencil",
              quantity: "2",
              price: "6"
            },
            {
              _id: 2,
              itemId: "M01",
              itemName: "Marker",
              quantity: "2",
              price: "50"
            },
            {
              _id: 3,
              itemId: "RB01",
              itemName: "Rubber",
              quantity: "2",
              price: "100"
            }
          ]
        },
        {
          id: 3,
          orderId: "010004",
          orderNumber: "010004",
          orderDate: "2019-09-01",
          items: [
            {
              _id: 0,
              itemId: "P01",
              itemName: "Pen",
              quantity: "2",
              price: "25"
            },
            {
              _id: 1,
              itemId: "PC01",
              itemName: "Pencil",
              quantity: "2",
              price: "6"
            },
            {
              _id: 2,
              itemId: "M01",
              itemName: "Marker",
              quantity: "2",
              price: "50"
            },
            {
              _id: 3,
              itemId: "D01",
              itemName: "Duster",
              quantity: "2",
              price: "100"
            }
          ]
        },
        {
          id: 4,
          orderId: "010005",
          orderNumber: "010005",
          orderDate: "2019-09-01",
          items: [
            {
              _id: 0,
              itemId: "PB01",
              itemName: "Plastic Bags",
              quantity: "2",
              price: "25"
            },
            {
              _id: 1,
              itemId: "PC01",
              itemName: "Pencil",
              quantity: "2",
              price: "6"
            },
            {
              _id: 2,
              itemId: "IN01",
              itemName: "Ink",
              quantity: "2",
              price: "50"
            },
            {
              _id: 3,
              itemId: "AS01",
              itemName: "A4 Sheet",
              quantity: "2",
              price: "100"
            }
          ]
        }
      ]
    }
  ]
};
