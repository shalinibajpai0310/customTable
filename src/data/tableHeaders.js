export const customerHeaders = [
  { title: "Customer Name", dataIndex: "name", key: "name" },
  { title: "Address", dataIndex: "address", key: "address" },
  { title: "Phonenumber", dataIndex: "phone", key: "phone" }
];

export const orderHeaders = [
  { title: "Order ID", dataIndex: "orderId", key: "orderId" },
  { title: "Order Number", dataIndex: "orderNumber", key: "orderNumber" },
  { title: "Order Date", dataIndex: "orderDate", key: "orderDate" }
];

export const itemHeaders = [
  { title: "Item Name", dataIndex: "itemName", key: "itemName" },
  { title: "Quantity", dataIndex: "quantity", key: "quantity" },
  { title: "Price", dataIndex: "price", key: "price" }
];
