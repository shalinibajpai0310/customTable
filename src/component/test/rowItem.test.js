import React from "react";
import { shallow,mount } from "enzyme";
import RowItem from "../rowItem";
import { customerHeaders,itemHeaders } from "../../data/tableHeaders";

describe("<RowItem />", () => {
    const item = {
        _id: "customer_0",
        index: 0,
        age: 31,
        name: "Price Valdez",
        email: "pricevaldez@metroz.com",
        phone: "+1 (804) 410-3749",
        address: "789 Remsen Avenue, Washington, Louisiana, 7638",
        orders: [],
        items:[]
      };
  it("renders without crashing", () => {
    const wrapper = shallow(
      <RowItem
        key={0}
        row={item}
        cols={customerHeaders}
        level={1}
        rowHighLight={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("simulate click event", () => {
    const wrapper = shallow(
      <RowItem
        key={0}
        row={item}
        cols={customerHeaders}
        level={1}
        rowHighLight={true}
      />
    );
    wrapper.find('button').simulate('click');
    expect(wrapper.state('showTable')).toEqual(true);
  });

  it("renders nested list", () => {
    const wrapper = shallow(
      <RowItem
        key={0}
        row={item}
        cols={itemHeaders}
        level={3}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
