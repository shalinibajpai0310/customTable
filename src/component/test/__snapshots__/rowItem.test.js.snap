// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowItem /> renders nested list 1`] = `ShallowWrapper {}`;

exports[`<RowItem /> renders nested list 1 1`] = `ShallowWrapper {}`;

exports[`<RowItem /> renders without crashing 1`] = `ShallowWrapper {}`;

exports[`<RowItem /> renders without crashing with level 1 1`] = `ShallowWrapper {}`;
